src/features/read/readSlice.ts

```diff
@@ -178,9 +179,51 @@ const selectQuestion = (state: RootState): IQuestion | null => {
 
 export const selectMode = (state: RootState): Mode | null => {
   const treatment = selectTreatment(state);
+  const book = selectBook(state)
+  const bookType = book?.fields.type;
+
```

Comment:
I am a big fan of "short-circuiting" functions. The goal is to get the majority of the code indented as little as possible. So you want your "if checks" around the shorter cases to handle.

In this case, handling pre- and post-tests is the long case to handle. 

So I personally would have re-arranged the logic to do:

```ts
// Deal with the normal case first
if (bookType !== "pre-test" && bookType !== "post-test") {
  if (treatment === "mixed") {
    return state.read.randomMode;
   }
   return treatment;
}

// Now all of your pre-/post-test logic can take up the rest of the function and not be as far indented for readability. 
```